pages/register.tsx

```diff
@@ -0,0 +1,251 @@
+            <form onSubmit={(e) => void handleRegister(e)}>
+              <TitleText className="text-center">Luo käyttäjätunnus</TitleText>
+              <div className="pt-5 pl-4 pr-4 flex flex-col">
+                <label>Etunimi</label>
+                <Input
+                  value={firstName}
+                  onChange={(e) => setFirstName(e.currentTarget.value)}
+                  className="border border-black"
+                  autoComplete="off"
+                  type="text"
+                  placeholder="Matti"
+                  name="firstName"
+                  spellCheck="false"
+                />
+                {isFirstNameError ? (
+                  <ErrorParagraph>{firstNameErrorText}</ErrorParagraph>
+                ) : null}
+
+                <label className="pt-5">Sukunimi</label>
+                <Input
+                  value={lastName}
+                  onChange={(e) => setLastName(e.currentTarget.value)}
+                  className="border border-black"
+                  autoComplete="off"
+                  type="text"
+                  placeholder="Meikäläinen"
+                  name="lastName"
+                  spellCheck="false"
+                />
+                {isLastNameError ? (
+                  <ErrorParagraph>{lastNameErrorText}</ErrorParagraph>
+                ) : null}
+
+                <label className="pt-5">Sähköposti</label>
+                <Input
+                  value={email}
+                  onChange={(e) => setEmail(e.currentTarget.value)}
+                  className="border border-black"
+                  autoComplete="off"
+                  type="text"
+                  placeholder="matti.meikalainen@email.com"
+                  name="email"
+                  spellCheck="false"
+                />
+                {isEmailError ? (
+                  <ErrorParagraph>{emailErrorText}</ErrorParagraph>
+                ) : null}
+
+                <label className="pt-5">Salasana</label>
+                <Input
+                  value={password}
+                  onChange={(e) => setPassword(e.currentTarget.value)}
+                  className="border border-black"
+                  autoComplete="off"
+                  type="password"
+                  placeholder="************"
+                  name="password"
+                />
+                {isPasswordError ? (
+                  <ErrorParagraph>{passwordErrorText}</ErrorParagraph>
+                ) : null}
+
+                <Button>Luo käyttäjätunnus</Button>
+                <p className="pt-6 text-xs text-gray-600">
+                  Onko sinulla jo tunnus?{' '}
+                  <span
+                    className="underline cursor-pointer hover:text-blue-500"
+                    onClick={() => {
+                      router.push('/login').catch((e) => console.error(e));
+                    }}
+                  >
+                    Kirjaudu sisään
+                  </span>
```

Comment:
Laitoin `router.push`-funktion suoraan `onClick`-funktioon, jotta ei tarvitse tehdä omaa handleria. Laitoin catchiin vain `console.error`:in. En saanut toistettua mitään virhettä kylläkään tuon `router.push`:n kanssa

Olisiko parempi laittaa `console.error`:n tilalle jokin "errorHandler"?

Tämä kysymys vastaa myös rivin 61 routeriin
https://github.com/samuliasmala/lahjalista/blob/09d97fb12f2ff412cc10983040965968619f71e3/pages/register.tsx#L61

Reply:
`console.error` on ihan hyvä, mutta käytä tässä mielummin linkkiä. Tätä ei voi esim. avata uuteen välilehteen mikä voisi joskus olla kätevää.